src/engine/Account-Imap.cpp

```diff
@@ -0,0 +1,912 @@
+{
+    std::string kvp_path;
+    std::string delim = "/";
+
+    if (!key) return;
+
+    ENTER(" ");
+
+    std::string str_key (key);
+
+    if (!category)
+        kvp_path = IMAP_FRAME + delim + str_key;
+    else
+    {
+        std::string str_category (category);
+        kvp_path = IMAP_FRAME + delim + str_category + delim + str_key;
+    }
+    xaccAccountBeginEdit (m_account);
+
+    if (qof_instance_has_slot (QOF_INSTANCE (m_account), kvp_path.c_str()))
+    {
+        qof_instance_slot_delete (QOF_INSTANCE (m_account), kvp_path.c_str());
+
+        if (category)
+        {
+            std::string str_category (category);
+            kvp_path = IMAP_FRAME + delim + str_category;
+
+            qof_instance_slot_delete_if_empty (QOF_INSTANCE (m_account), kvp_path.c_str());
+        }
+        qof_instance_slot_delete_if_empty (QOF_INSTANCE (m_account), IMAP_FRAME);
+    }
+    qof_instance_set_dirty (QOF_INSTANCE (m_account));
+    xaccAccountCommitEdit (m_account);
+
+    LEAVE(" ");
+}
+
+
+using AccountTokenCount = std::pair<std::string, int64_t>; // guid, token_count
+//using AccountTokenCountPtr = std::shared_ptr<AccountTokenCount>;
+//using TokenAccountsInfo = std::pair<std::vector<AccountTokenCountPtr>, int64_t>;
+
+/** total_token_count and the token_count for a given account let us calculate the
+ * probability of a given account with any single token
+ */
+struct TokenAccountsInfo
+{
+    std::vector<AccountTokenCount> accounts; // vector of AccountTokenCount
+    int64_t total_token_count; // the sum of token_count for a given token
+};
+
+/** gpointer is a pointer to a struct TokenAccountsInfo
+ * \note Can always assume that keys are unique, reduces code in this function
+ */
+static void
+buildTokenInfo(const char *key, const GValue *value, gpointer data)
+{
+    struct TokenAccountsInfo *tokenInfo = (struct TokenAccountsInfo*)data;
+
+    //  PINFO("buildTokenInfo: account '%s', token_count: '%" G_GINT64_FORMAT "'", (char*)key,
+    //                  g_value_get_int64(value));
+
+    /* add the count to the total_token_count */
+    tokenInfo->total_token_count += g_value_get_int64(value);
+
+    /* fill in the account guid and number of tokens found for this account */
+    AccountTokenCount account_token_count = std::make_pair (key, g_value_get_int64(value));
+
+    /* append onto the vector the new AccountTokenCount pair */
+    tokenInfo->accounts.push_back (account_token_count);
+}
+
+/** intermediate values used to calculate the bayes probability of a given
+    account where p(AB) = (a*b)/[a*b + (1-a)(1-b)], product is (a*b),
+    product_difference is (1-a) * (1-b)
+ */
+class AccountProbability
+{
+    public:
+    double m_product;             /* product of probabilities */
+    double m_product_difference;  /* product of (1-probabilities) */
+
+    AccountProbability (double p, double pd)
+    {
+        m_product = p;
+        m_product_difference = pd;
+    }
+};
```

Comment:
say
```
    AccountProbability (double p, double pd) : m_product{p}, m_product_difference{pd}{};
```
Otherwise you're initializing and setting instead of just initializing.